### PR TITLE
Adaptive scope reduction and max eval option for covsearch

### PR DESCRIPTION
**Pull request by JohanWBorg**
Possibility of setting the adaptive scope reduction (from scm+) and max eval when running covsearch.
Should implement #2881 as well as #2880

### CODE REVIEW

File: docs/covsearch.rst [hunk 1]
```diff
@@ -82,6 +82,14 @@ Optional
 | ``algorithm``                               | The search :ref:`algorithm<algorithm_covsearch>` to use               |
 |                                             | (default is `'scm-forward-then-backward'`)                            |
 +---------------------------------------------+-----------------------------------------------------------------------+
+| ``max_eval``                                | Limit the number of function evaluations to 3.1 times that of the     |
+|                                             | base model. Default is False.                                         |
++---------------------------------------------+-----------------------------------------------------------------------+
+| ``adaptive_scope_reduction``                | Stash all non-significant parameter-covariate effects to be tested    |
```
Comment (rikardn) [inline docs/covsearch.rst:88]: I think this sentence is slightly confusing. Perhaps: "Stash all non-significant parameter-covariate effects at each step in the forward search for later evaluation. As soon as all significant effects have been tested the stashed effects gets evaluated in a normal forward approach."

File: src/pharmpy/tools/covsearch/tool.py [hunk 2]
```diff
@@ -465,12 +600,16 @@ def _greedy_search(
...
         }
 
-    return SearchState(
-        state.user_input_modelentry,
-        state.start_modelentry,
-        best_candidate_so_far,
-        all_candidates_so_far,
-    )
+        # Filter away any stashed effects as well
+        insig_param_cov_eff = tuple((eff[0], eff[1]) for eff in insignificant_effects.keys())
```
Comment (rikardn) [inline src/pharmpy/tools/covsearch/tool.py:604]: Use the term word `nonsignificant` rather than `insignificant`.

**Comment by JohanWBorg** [comment]
Sorry for the delay, the requested changes have now been fixed!

**Comment by rikardn** [comment]
Thanks!

**Pull request merged**